src/Sylius/Bundle/CoreBundle/Form/Type/CartItemType.php

```diff
@@ -34,7 +34,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
     {
         parent::buildForm($builder, $options);
 
-        if (isset($options['product']) && $options['product']->hasVariants()) {
+        if (isset($options['product']) && $options['product']->hasVariants() && !$options['product']->isSimple()) {
```

Comment:
With regards to commit squashing, can you not do it when merging? as per this blog post - https://github.com/blog/2141-squash-your-commits
I'm not sure how to do it yet, but i'll give it a try